### ISSUE TITLE
Error while having the Interleaved table having only one primary key which is same as the parent table

### ISSUE THREAD

**Issue by rajakumare1** [GoogleCloudPlatform/spring-cloud-gcp#1646]
**Describe the bug**
com.google.cloud.spring.data.spanner.core.mapping.SpannerDataException: A child table (tablename) must contain the primary key columns of its parent (tablename) in the same order starting the first column with additional key columns after.

If you have Interleave table that has only the same primary key as the Parent and doesnt have any additional key, then getting this Exception

Note: I have business reason to logically have another table to store separate information from ParentTable

Sample

public class Parent {

    @PrimaryKey(keyOrder = 1)
    @Column(name = "ParentId", nullable = false)
    private String ParentId;

  @Column(name = "details", nullable = false)
private String details

    @Interleaved
    private List<ParentLogical> parentLogical;

}

public class ParentLogical {

    @PrimaryKey(keyOrder = 1)
    @Column(name = "ParentId", nullable = false)
    private String ParentId;

  @Column(name = "associatedlogicalName", nullable = false)
private String associatedlogicalName

}

**Comment by meltsufin** [comment]
Does the Spanner database even allow this? I believe the child table needs a secondary key. Have you tried just adding `  @PrimaryKey(keyOrder = 2)` for `associatedlogicalName`?

**Comment by rajakumare1** [comment]
Yes it allows the users to create interleave without the 2nd key order . I am 100% sure since I have created it and that’s when I ran into the issue

**Comment by meltsufin** [comment]
I guess we just don't support this edge case. Have you tried my suggestion in the previous comment?

**Comment by rajakumare1** [comment]
If I recreate the table ParentLogical and make a primary key with both the columns in the table then it works.  

But my use case is totally different. The secondary table is just a logical table created so that main table doesnt have too many columns. 

 If I have the same primary key in both table, it doesnt work. The Database supports it so I humbly think the Spring Data should support too.

**Comment by meltsufin** [comment]
Understood, I'll leave this issue in as a feature request.

**Comment by NicolaeV** [comment]
I'm commenting to express my desire for this existing Spanner behavior to be represent-able using Spring Cloud Data Spanner.
